golangci-lint fails to build, aborts CI

Since Apr 23, 2020 CI runs have been failing with variations of these errors:
```
go get -u golang.org/x/tools/cmd/goyacc \
	github.com/golangci/golangci-lint/cmd/golangci-lint \
	github.com/dvyukov/go-fuzz/go-fuzz-build
# github.com/quasilyte/go-ruleguard/ruleguard/typematch
../../quasilyte/go-ruleguard/ruleguard/typematch/typematch.go:74:15: undefined: strings.ReplaceAll
Makefile:332: recipe for target 'install_prerequisites' failed
make: *** [install_prerequisites] Error 2
The command "make install_prerequisites" exited with 2.
...
make[1]: golangci-lint: Command not found
```

```
go get -u golang.org/x/tools/cmd/goyacc \
	github.com/golangci/golangci-lint/cmd/golangci-lint \
	github.com/dvyukov/go-fuzz/go-fuzz-build
# github.com/golangci/golangci-lint/pkg/golinters
../../golangci/golangci-lint/pkg/golinters/golint.go:21:14: l.LintPkg undefined (type *"github.com/golangci/lint-1".Linter has no field or method LintPkg)
../../golangci/golangci-lint/pkg/golinters/gomnd.go:13:6: cannot use magic_numbers.Analyzer (type *"github.com/tommy-muehle/go-mnd/vendor/golang.org/x/tools/go/analysis".Analyzer) as type *"golang.org/x/tools/go/analysis".Analyzer in slice literal
Makefile:332: recipe for target 'install_prerequisites' failed
```

--- a/Makefile
+++ b/Makefile
@@ -339,8 +339,14 @@ install_prerequisites:
 	sudo apt-get install -y -q g++-mips64el-linux-gnuabi64 || true
 	sudo apt-get install -y -q ragel clang-format
 	go get -u golang.org/x/tools/cmd/goyacc \
-		github.com/golangci/golangci-lint/cmd/golangci-lint \
 		github.com/dvyukov/go-fuzz/go-fuzz-build
+	# Runs golangci-lint when go is new enough. Old versions get a
+	# free pass (except on CI which has newer go version).
+	if [ "$$(go version)" \> "go version go1.12" ]; then \
+		GO111MODULE=on go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.25.0 ; \
+	else \
+		ln -sf /bin/true $$(go env GOPATH)/bin/golangci-lint ; \
+	fi
 
 check_copyright:
 	./tools/check-copyright.sh

